src/tactic/polyrith.lean

```diff
@@ -45,6 +45,15 @@ given to the `linear_combination` tactic, which completes the process by checkin
 
 open tactic native
 
+meta def tactic.get_mathlib_root : tactic string :=
+do
+  e ← tactic.get_env,
+  some p ← pure (e.decl_olean `tactic.get_mathlib_root) |
+    fail!"Cannot call tactic.get_mathlib_src from this file",
+  ("polyrith.lean" :: "tactic" :: "src" :: parts) ←
+    pure (p.split $ (= '/')).reverse | fail!"Bad path: {p}",
+  pure (string.intercalate "/" parts.reverse)
+
```

Comment:
This should be somewhere much more basic; any suggestions?

Reply:
Probably next to `get_mathlib_dir` (https://github.com/leanprover-community/mathlib/blob/861589f224456f278c61596c0ee8a1ba449a6e8b/src/tactic/core.lean#L2259)? Is this equivalent to `(λ s, string.popn_back s 5) <$> tactic.get_mathlib_dir` (modulo error reporting)?

Reply:
Ah, I didn't know that existed. Splitting /s feels cleaner than counting characters, but certainly we shouldn't have two versions of this tactic.

I think probably `(λ s, s + "/..") <$> tactic.get_mathlib_dir` is best, which maybe has too many or too few slashes

Reply:
Replaced (no slashes needed), it seems to work!